src/main/java/org/aksw/word2vecrestful/word2vec/Word2VecModel.java

```diff
@@ -74,6 +79,36 @@ private void fetchVectorDimensions() {
 	  resMap.put(minWord, minVec);
 	  return resMap;
   }
+  
+  /**
+   * Method to fetch a single closest word entry to the passed input vector 
+   * inside the subset of word2vec model
+   * @param inpvec - input vector to find closest word for
+   * @param subsetKey - key to identify the subset model
+   * @return Map of the single closest word and its vector
+ * @throws IOException 
+ * @throws FileNotFoundException 
+ * @throws JsonProcessingException 
+   */
+  public Map<String, float[]> getClosestEntryInSub(float[] inpvec, String subsetKey) throws JsonProcessingException, FileNotFoundException, IOException{
+	  Map<String, float[]> resMap = new HashMap<>();
+	  double minDist = -2;
+	  String minWord = null;
+	  float[] minVec = null;
+	  double tempDist;
+	  // Loop on the subset
+	  for(String word : DataSubsetProvider.fetchSubsetWords(subsetKey)) {
```

Comment:
This looks like you have to load the subset from file for every single request. Wouldn't it be better to have an instance of the DataSubsetProvider class that is able to simply store the loaded list in memory? As we already discussed, this wouldn't consume a lot of memory since the Strings are already loaded in the JVM because of the in-memory Word2Vec model.